src/main/kotlin/org/rust/lang/refactoring/extractFunction/RsExtractFunctionDialog.kt

```diff
@@ -0,0 +1,64 @@
+package org.rust.lang.refactoring.extractFunction
+
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.DialogWrapper
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import com.intellij.refactoring.ui.NameSuggestionsField
+import com.intellij.ui.components.dialog
+import org.rust.lang.core.psi.RsFunction
+import java.awt.Dimension
+import java.awt.GridLayout
+import javax.swing.JComboBox
+import javax.swing.JLabel
+import javax.swing.JPanel
+
+fun extractFunctionDialog(
+    project: Project,
+    file: PsiFile,
+    config: RsExtractFunctionConfig
+) {
+    val contentPane = JPanel(GridLayout(2, 2))
+    val functionNameLabel = JLabel()
+    val functionNameField = NameSuggestionsField(project)
+    val visibilityLabel = JLabel()
+    val visibilityBox = JComboBox<String>()
+
+    contentPane.size = Dimension(522, 396)
+    functionNameLabel.text = "Name:"
+
+    visibilityLabel.text = "Visibility:"
+    with(visibilityBox) {
+        addItem("Public")
+        addItem("Private")
+    }
+    visibilityBox.selectedItem = "Private"
+
+    with(contentPane) {
+        add(visibilityLabel)
+        add(functionNameLabel)
+        add(visibilityBox)
+        add(functionNameField)
+    }
+    config.name = functionNameField.enteredName
+    config.visibilityLevelPublic = visibilityBox.selectedItem == "Public"
+
+    dialog(
+        "Extract Function",
+        contentPane,
+        false,
+        functionNameField,
+        true,
+        project,
+        null,
+        null,
```

Comment:
In general, it's usually better to use named arguments when passing literal constants like `null` or `false` to enhance readability.